apps/api/src/purchase-orders/purchase-orders.service.ts

```diff
@@ -0,0 +1,202 @@
+import { Injectable, NotFoundException, Logger } from '@nestjs/common';
+import { PurchaseOrders } from "@prisma/client";
+import { CreatePurchaseOrderDto } from './dto/create-purchase-order.dto';
+import { UpdatePurchaseOrderDto } from './dto/update-purchase-order.dto';
+import { PrismaService } from "../prisma.service";
+
+@Injectable()
+export class PurchaseOrdersService {
+  constructor(private prisma: PrismaService) {
+  }
+
+  // Just a note that the naming convention here seems inconsistent. In prisma, PurchaseOrders is plural while ParentItem is singular
+  async findAll(sortField: string, sortOrder: string): Promise<PurchaseOrders[]> {
+
+    const purchaseOrders = await this.prisma.purchaseOrders.findMany({
+      /* 
+        There is a problem here that should be noted between Prisma and SQLite specifically for dates. Prisma maps DateTime fields to a numeric in SQLite. 
+        The seed data stored the dates in ISO format but when creating or update records, it will be stored as an integer which will make it seem like the sort is broken. 
+        For more info, see: https://github.com/prisma/prisma/issues/8510
+      */
+      orderBy: {
+        [sortField]: sortOrder,
+      },
+      include: {
+        purchase_order_line_items: true,
+        goods_receipts: true,
+        supplier_invoices: true,
+      },
+    });
+
+    return purchaseOrders.map(order => ({
+      ...order,
+      total_quantity: order.purchase_order_line_items.reduce((sum, item) => sum + item.quantity, 0),
+      total_cost: order.purchase_order_line_items.reduce((sum, item) => sum + item.quantity * Number(item.unit_cost), 0),
+    }));
+  }
+  
+  async create(createPurchaseOrderDto: CreatePurchaseOrderDto): Promise<PurchaseOrders> {
+    return this.prisma.purchaseOrders.create({
+      data: {
+        vendor_name: createPurchaseOrderDto.vendor_name,
+        order_date: createPurchaseOrderDto.order_date,
+        expected_delivery_date: createPurchaseOrderDto.expected_delivery_date,
+        purchase_order_line_items: {
+          create: createPurchaseOrderDto.purchase_order_line_items.map(item => ({
+            item_id: item.item_id,
+            quantity: item.quantity,
+            unit_cost: item.unit_cost,
+          })),
```

Comment:
Looks like most of you transformations are inline, do you like this pattern or do you ever use a separate transformer function or class?

Reply:
Whats the benefits of either approach, inline or imported?

Reply:
In cases where you can simply take what you have in the DTO (along with the transformations you get from the transform decorator) and map it to the database object in a straightforward manner, I prefer having transformations inline. If the transformation is more complex, a separate transformer function makes sense. 

The benefits of the inline approach are simplicity and a slight increase in performance while the benefits of the imported approach are it's DRY, it is more maintainable for complex transformations, and it is easier to unit test the transformation logic separately from the business logic.